api/src/main/java/io/minio/MinioClient.java

```diff
@@ -899,9 +899,14 @@ private Request createRequest(Method method, String bucketName, String objectNam
     }
 
     if (objectName != null) {
-      // Limitation: OkHttp does not allow to add '.' and '..' as path segment.
-      for (String pathSegment : objectName.split("/")) {
-        urlBuilder.addEncodedPathSegment(S3Escaper.encode(pathSegment));
+      if (!objectName.equals("/")) {
+        // Limitation: OkHttp does not allow to add '.' and '..' as path segment.
+        for (String pathSegment : objectName.split("/")) {
+          urlBuilder.addEncodedPathSegment(S3Escaper.encode(pathSegment));
+        }
+      } else if (objectName.equals("/")) {
```

Comment:
Have this `if` checking `.endsWith("/")` and add comment about this fix and issue#